src/hmm/hmm-topology.h

```diff
@@ -1,194 +0,0 @@
-// MERCHANTABLITY OR NON-INFRINGEMENT.
-// See the Apache 2 License for the specific language governing permissions and
-// limitations under the License.
-
-#ifndef KALDI_HMM_HMM_TOPOLOGY_H_
-#define KALDI_HMM_HMM_TOPOLOGY_H_
-
-#include "base/kaldi-common.h"
-#include "util/const-integer-set.h"
-
-
-namespace kaldi {
-
-
-/// \addtogroup hmm_group
-/// @{
-
-/*
- // The following would be the text form for the "normal" HMM topology.
- // Note that the first state is the start state, and the final state,
- // which must have no output transitions and must be nonemitting, has
- // an exit probability of one (no other state can have nonzero exit
- // probability; you can treat the transition probability to the final
- // state as an exit probability).
- // Note also that it's valid to omit the "<PdfClass>" entry of the <State>, which
- // will mean we won't have a pdf on that state [non-emitting state].  This is equivalent
- // to setting the <PdfClass> to -1.  We do this normally just for the final state.
- // The Topology object can have multiple <TopologyEntry> blocks.
- // This is useful if there are multiple types of topology in the system.
-
- <Topology>
- <TopologyEntry>
- <ForPhones> 1 2 3 4 5 6 7 8 </ForPhones>
- <State> 0 <PdfClass> 0
- <Transition> 0 0.5
- <Transition> 1 0.5
- </State>
- <State> 1 <PdfClass> 1
- <Transition> 1 0.5
- <Transition> 2 0.5
- </State>
- <State> 2 <PdfClass> 2
- <Transition> 2 0.5
- <Transition> 3 0.5
- <Final> 0.5
- </State>
- <State> 3
- </State>
- </TopologyEntry>
- </Topology>
-*/
-
-// kNoPdf is used where pdf_class or pdf would be used, to indicate,
-// none is there.  Mainly useful in skippable models, but also used
-// for end states.
-// A caveat with nonemitting states is that their out-transitions
-// are not trainable, due to technical issues with the way
-// we decided to accumulate the stats.  Any transitions arising from (*)
-// HMM states with "kNoPdf" as the label are second-class transitions,
-// They do not have "transition-states" or "transition-ids" associated
-// with them.  They are used to create the FST version of the
-// HMMs, where they lead to epsilon arcs.
-// (*) "arising from" is a bit of a technical term here, due to the way
-// (if reorder == true), we put the transition-id associated with the
-// outward arcs of the state, on the input transition to the state.
-
-/// A constant used in the HmmTopology class as the \ref pdf_class "pdf-class"
-/// kNoPdf, which is used when a HMM-state is nonemitting (has no associated
-/// PDF).
-
-static const int32 kNoPdf = -1;
-
-/// A class for storing topology information for phones.  See  \ref hmm for context.
-/// This object is sometimes accessed in a file by itself, but more often
-/// as a class member of the Transition class (this is for convenience to reduce
-/// the number of files programs have to access).
-
-class HmmTopology {
- public:
-  /// A structure defined inside HmmTopology to represent a HMM state.
-  struct HmmState {
```

Comment:
@danpovey HmmState is still used in topology.cc. If you have time soon, could you fix that part of the build? I can look into it, but I'm not entirely sure what all has happened there...

Reply:
It's delaying my work on creating a cmake-based build.

Reply:
[I dont trust mail to github, it's been lost recently, so resending], @hhadian  was going to look at that when he gets time.  I drafted the interface but didn't do the implementation.